OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -192,14 +184,14 @@ public void PrerequisitesItemVisible(string key)
 			producible[self.World.Map.Rules.Actors[key]].Visible = true;
 		}
 
-		public ProductionItem CurrentItem()
+		public virtual bool IsProducing(ProductionItem item)
 		{
-			return queue.ElementAtOrDefault(0);
+			return Queue.Count > 0 && Queue[0] == item;
 		}
 
 		public IEnumerable<ProductionItem> AllQueued()
```

Comment:
That's probably reasonable in this case.

In the general case it's often a problem. Exposing interfaces or wrapper collections instead of concrete collections kills performance quite a lot, so I wouldn't go mass applying it to things, despite how nice it would be for better semantics.